internal/profiles/util.go

```diff
@@ -299,6 +301,59 @@ func MergeDatabaseGetByNameIntoProfiles(ppl []db.GetProfileByProjectAndNameRow)
 	return profiles
 }
 
+// Derive the profile name from the profile display name
+func DeriveProfileNameFromDisplayName(
+	profile *pb.Profile,
+) (profileName string) {
+
+	displayName := profile.GetDisplayName()
+	name := profile.GetName()
+
+	if displayName != "" && name != "" {
+		// when both a display name and a profile name are provided
+		// then the profile name from the incoming request is used as the profile name
+		profileName = name
+
+		// when both a display name and a profile name are provided, but the project already has a profile with that name
+
+	} else if displayName != "" && name == "" {
+		// when a display name is provided, but no name
+		// then the profile name is created and saved based on the profile display name
+
+		profileName = CleanDisplayName(displayName)
+
+	}
+
+	return profileName
+}
+
+// The profile name should be derived from the profile display name given the following logic
+func CleanDisplayName(displayName string) string {
```

Comment:
this function probably doesn't have to be exported

Reply:
Thank you, @jhrozek. Fixed now.